AutoCollection/CorrelationContextManager.ts

```diff
@@ -168,7 +168,25 @@ export class CorrelationContextManager {
                 return AppInsightsAsyncCorrelatedErrorWrapper.apply(Object.create(AppInsightsAsyncCorrelatedErrorWrapper.prototype), arguments);
             }
 
+            // Is this object set to rewrite the stack?
+            // If so, we should turn off some Zone stuff that is prone to break
+            var stackRewrite = orig['stackRewrite'];
+            if (orig['prepareStackTrace']) {
+                orig['stackRewrite'] = false;
+                var stackTrace = orig['prepareStackTrace'];
+                orig['prepareStackTrace'] = (e, s) => {
+                    // Remove some AI and Zone methods from the stack trace
+                    // Otherwise we leave side-effects
+                    s.splice(0, 3);
```

Comment:
3 here is the number of methods we add to the stack trace. The library that we crash on will still fail if we don't remove ourselves from the trace